### ISSUE TITLE
<DialogDisclosure as="div"> triggers additional event on focused Dialog element

### ISSUE THREAD

**Issue by folz** [reakit/reakit#598]
## 🐛 Bug report

When rendering a `<DialogDisclosure as="div">`, activating it with the Space key seems to cause an event on the first element focused in the Dialog. This does not occur when rendering `<DialogDisclosure as="button">`, or when using activating the `as="div"` with Enter or with a mouse click.

### Current behavior

See above.

### Steps to reproduce the bug

Provide a repo or sandbox with the bug and describe the steps to reproduce it.

1. Open sandbox: https://codesandbox.io/s/elastic-frog-wq14j
2. Focus the DialogDisclosure and press Space
3. The Dialog will open and incorrectly immediately close

Variants:

Hold down Space, then click elsewhere in the Dialog to deselect the close button. Release Space once the close button is not focused. The Dialog correctly does not close.

Use Enter to activate the `<DialogDisclosure as="div">`. The Dialog correctly does not close.

Remove `as="div"`, so the rendered element is `<button>`. Activate DialogDisclosure with Space. The Dialog correctly does not close.

### Expected behavior

The DialogDisclosure should behave the same when rendering `"div"` as it does when rendering `"button"`.

### Possible solutions

¯\\_(ツ)_/¯

### Environment

See dependencies in the codesandbox.

I'm on Mac OS Catalina.

This behavior slightly differs between Chrome/Safari and Firefox. In Firefox, the behavior always occurs. With Chrome and Safari, you have to hold down space for 100 or so ms until the first Dialog element is focused - if you release too early, all works as expected and the Dialog focused element is not triggered. I'm not sure why. Let me know if you have trouble reproducing and I can make a video.

**Comment by folz** [comment]
To clarify why this is an issue for me, and why I would prefer to keep using `as="div"` instead of using button: our design system supports making any component look like any other component, with an API like

```js
<DialogDisclosure variant="Button.primary" ...>
<Touchable variant="Card" ...>
```

It supports this by assuming all components are `<div>s` with the correct interaction handlers (e.g. `tabIndex`, supporting enter/space), and providing styles to turn any unstyled div into the correct look.

Because `<button>` comes with built-in browser styles, and we choose to not ship with CSS resets, I've re-exported all button-default reakit components to use `as: "div"` instead. So far this has worked seamlessly, thanks to Tabbable's great support for making non-`<button>` elements support accessible interaction.

It would be ideal to fix this bug, rather than having to go back and change all styles to handle div-or-button. If you don't have bandwidth to look at this right now but can think of good places to start, please let me know and I'm happy to do some digging.

**Comment by diegohaz** [comment]
Thanks for reporting it, @folz! I think it has been fixed by #596

See the CodeSandbox created by that PR: https://codesandbox.io/s/morning-butterfly-xv48l

But since that also introduces a breaking change, that will have to wait a little bit to get released.

The issue is probably that on native buttons, <kbd>Space</kbd> only triggers the button on key up. But the current `Tabbable`'s  behavior on master does that on key down. With the new `Clickable` component, this behavior is corrected.

**Comment by folz** [comment]
I think you're right - I'm not able to trigger that behavior anymore in the sandbox you linked. I look forward to the release!

**Comment by diegohaz** [comment]
Should be fixed on v1.0.0-rc.0
